spec/system/placements/schools/partner_providers/add_a_provider_spec.rb

```diff
@@ -0,0 +1,262 @@
+    and_i_click_on_confirm_and_add_provider
+    then_i_return_to_the_provider_index
+    and_i_see_ogdenville_university_listed
+    and_i_see_success_message_for_ogdenville
+    but_a_notification_email_is_not_sent_to_ogdenville_university
+
+    then_i_click_on_add_provider
+    and_i_enter_springfield_university
+    then_i_see_a_dropdown_item_for_springfield_university
+    and_i_click_on_the_springfield_university_dropdown_item
+    and_i_click_on_continue
+    then_i_see_confirm_provider_details_page_for_springfield_university
+
+    # User reconsiders selecting a provider using back link
+    when_i_click_on_back
+    then_i_see_the_search_input_prefilled_with_springfield_university
+    and_i_click_on_continue
+    then_i_see_confirm_provider_details_page_for_springfield_university
+
+    # User reconsiders selecting a provider using change link
+    when_i_click_on_change
+    then_i_see_the_search_input_prefilled_with_springfield_university
+    and_i_click_on_continue
+    then_i_see_confirm_provider_details_page_for_springfield_university
+
+    and_i_click_on_confirm_and_add_provider
+    then_i_return_to_the_provider_index
+    and_i_see_springfield_university_listed
+    and_i_see_success_message_for_springfield_university
+    and_a_notification_email_is_sent_to_springfield_university
+  end
+
+  private
+
+  def given_provider_exists
+    @springfield_university = create(:placements_provider,
+                                     name: "Springfield University",
+                                     ukprn: "10101010",
+                                     urn: "101010",
+                                     email_address: "reception@springfield.ac.uk",
+                                     telephone: "0101 010 0101",
+                                     website: "http://www.springfield.ac.uk",
+                                     address1: "Undisclosed")
+    @springfield_university_user = create(:placements_user, providers: [@springfield_university])
+
+    @ogdenville_university = create(:placements_provider,
+                                    name: "Ogdenville University",
+                                    ukprn: "00100010",
+                                    urn: "000010",
+                                    email_address: "reception@ogdenville.ac.uk",
+                                    telephone: "0000 000 1111",
+                                    website: "http://www.ogdenville.ac.uk",
+                                    address1: "1 Main Street")
+    @ogdenville_university_user = create(:placements_user, providers: [@ogdenville_university])
+
+    @shelbyville_university = create(:placements_provider, name: "Shelbyville University")
+  end
+
+  def and_i_am_signed_in
+    @school = create(:school, :placements)
+    given_i_am_signed_in_as_a_placements_user(organisations: [@school])
+  end
+
+  def when_i_am_on_the_providers_index_page
+    visit placements_school_partner_providers_path(@school)
```

Comment:
I believe we're trying to avoid using `visit` in the new system spec format. Try clicking `Providers` in the nav bar instead.